backend/src/test/java/com/example/demo/KanbanServiceTest.java

```diff
@@ -0,0 +1,110 @@
+package com.example.demo;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+class KanbanServiceTest {
+
+    @Test
+    void shouldSucceedOnAddMethodCall() {
+        Item item = new Item("Projekt1", "Beschreibung Projekt 1", StatusEnum.OPEN);
+
+        KanbanProjectRepo kanbanProjectRepo = Mockito.mock(KanbanProjectRepo.class);
+
+        KanbanService kanbanService = new KanbanService(kanbanProjectRepo);
+        kanbanService.addItem(item);
+
+        Mockito.verify(kanbanProjectRepo).addItem(item);
+    }
+
+    @Test
+    void shouldReturnCollectionOfAllItems() {
+        Item item1 = new Item("Projekt1", "Beschreibung Projekt 1", StatusEnum.OPEN);
+        Item item2 = new Item("Projekt2", "Beschreibung Projekt 2", StatusEnum.OPEN);
+
+        KanbanProjectRepo kanbanProjectRepo = Mockito.mock(KanbanProjectRepo.class);
+        Mockito.when(kanbanProjectRepo.getAllItems())
+                .thenReturn(List.of(item1, item2));
+
+        KanbanService kanbanService = new KanbanService(kanbanProjectRepo);
+        Assertions.assertThat(kanbanService.getAllItems())
+                .isUnmodifiable()
+                .contains(item1, item2);
+    }
+
+    @Test
+    void shouldReturnCorrectItemStatusAfterMovedToNext() {
+        Item item = new Item("Projekt1", "Beschreibung Projekt 1", StatusEnum.OPEN);
+
+        KanbanProjectRepo kanbanProjectRepo = Mockito.mock(KanbanProjectRepo.class);
+
+        KanbanService kanbanService = new KanbanService(kanbanProjectRepo);
+        try {
```

Comment:
Ich würde hier (und in den anderen Tests) kein try-catch machen. Wenn die Exception fliegt, schlägt dein Test fehl und das ist auch gut so.
Wenn deine `moveToNext`-Methode so aussieht
```java
public Item moveToNext(Item item){
    StatusEnum newStatus = item.getStatus().next();
    item.setStatus(newStatus);
    Item item = kanbanProjectRepo.replaceItem(item).orElseThrow();
    throw new NoSuchElementException();
    return item;
}
```
wäre der Test weiterhin grün. Das Beispiel ist zwar ein wenig konstruiert, zeigt aber warum in dem Test das try-catch zum Problem werden kann.